### PR TITLE
Implement combine_similar for timeseries

### PR DESCRIPTION
**Pull request by phofl**
@rjzamora I'd appreciate thoughts here when you get back

This isn't super important for users, but not having combine_similar here makes most of our benchmarks meaningless since we'll end up doing a lot of unnecessary work.

### CODE REVIEW

**Comment by rjzamora** [comment]
Planning to look carefully at this today. My initial question is "can we use `_absorb_projections = True` to avoid adding new `_combine_similar` logic?". Perhaps `Timeseries` is somehow different from the other `BlockwiseIO` cases?

**Comment by rjzamora** [comment]
>Timeseries changes dtypes instead of columns if a projection is pushed down

Yeah, that makes sense. We do want the user to pass `dtypes` to `timeseries` (and not specify `columns`). However, I don't really see the harm in (privately) supporting `columns` within `Timeseries`. In other words, we can probably avoid code duplication if we just add a `dtypes` property that simply checks `self.operand("columns")` to apply a projection.

**Comment by phofl** [comment]
Yep you are correct, that makes more sense. I've adjusted this pr accordingly

File: dask_expr/datasets.py [hunk 1]
```diff
@@ -53,42 +70,45 @@ def random_state(self):
...
-        if isinstance(parent, Projection) and len(self.dtypes) > 1:
-            dtypes = {col: self.operand("dtypes")[col] for col in parent.columns}
-            out = Timeseries(
+        if isinstance(parent, Projection) and is_dataframe_like(self._meta):
+            make_series = is_scalar(parent.operand("columns")) and not self._series
+            return Timeseries(
                 self.start,
                 self.end,
-                dtypes=dtypes,
+                dtypes=self.operand("dtypes"),
```
Comment (rjzamora) [inline dask_expr/datasets.py:102]: Now that we aren't changing `dtypes` here, then we can remove `Timeseries._simplify_up` altogether. We also don't need the `_token_dtypes` operand anymore.

File: dask_expr/datasets.py [hunk 2]
```diff
@@ -31,19 +38,29 @@ class Timeseries(PartitionsFiltered, BlockwiseIO):
...
+            return result[result.columns[0]]
+        return result
 
     def _divisions(self):
         return pd.date_range(start=self.start, end=self.end, freq=self.partition_freq)
 
+    @property
+    def _dtypes(self):
+        dtypes = self.operand("dtypes")
+        return {col: dtypes[col] for col in self.operand("columns")}
```
Comment (rjzamora) [inline dask_expr/datasets.py:62]: What if `self.operand("columns")` is `None`?

File: dask_expr/datasets.py (hunk 1 above)
Comment (phofl) [reply dask_expr/datasets.py]: Yeah that's fair.

We still need the token. We have to be able to distinguish between different time series calls

```
df = timeseries()
df2 = timeseries()

(df + df2) != 2*df
```

File: dask_expr/datasets.py (hunk 2 above)
Comment (phofl) [reply dask_expr/datasets.py:60]: Good point, should not happen. Adjusted the signature

File: dask_expr/datasets.py (hunk 1 above)
Comment (rjzamora) [reply dask_expr/datasets.py]: >We have to be able to distinguish between different time series calls

Wont the seeds be different (making them distinguishable anyway)?

Comment (phofl) [reply dask_expr/datasets.py]: I have another PR to remove the seed, specifying a seed makes serialising the graph very slow (should probably optimise this, but didn't have time yet) and is inconsistent with dask/dask, which is what makes benchmarking hard

**Review by rjzamora — APPROVED** [review]

File: dask_expr/datasets.py (hunk 1 above)
Comment (phofl) [reply dask_expr/datasets.py]: We should be good here though, the randint call is only triggered when constructing the graph, so not when we go through optimize

**Pull request merged**